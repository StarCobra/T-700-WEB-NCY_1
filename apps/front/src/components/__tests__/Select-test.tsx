import React from "react";
import { act, create } from "react-test-renderer";
import Component from "../Content";

describe("Components > Content", () => {
  const tests = [
    {
      description: "without props",
      props: {},
    },
    {
      description: "with type unique",
      props: {
        label: "Lorem Ipsum",
        options: [
          { value: "BTC", label: "Bitcoin" },
          { value: "ETH", label: "Ethereum" },
          { value: "LTC", label: "Litecoin" },
        ],
      },
    },
    {
      description: "with type multiple",
      props: {
        label: "Lorem Ipsum",
        type: "multiple",
        options: [
          { value: "BTC", label: "Bitcoin" },
          { value: "ETH", label: "Ethereum" },
          { value: "LTC", label: "Litecoin" },
        ],
      },
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;

      await act(async () => {
        root = create(<Component {...test.props} />);
      });

      expect(root).toMatchSnapshot();
    });
  });
});
