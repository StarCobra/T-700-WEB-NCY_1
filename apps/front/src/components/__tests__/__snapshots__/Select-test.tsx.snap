// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Content should render the component with type multiple 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
/>
`;

exports[`Components > Content should render the component with type unique 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
/>
`;

exports[`Components > Content should render the component without props 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
/>
`;
